tests/general/base.py

```diff
@@ -0,0 +1,200 @@
+        self.geo2 = self.root.add_child(code="EC", level="province")
+        self.dc1 = self.geo1.add_child(code="DC1", level="muni")
+        self.create_boundary(self.root, self.version)
+        self.create_boundary(self.geo1, self.version)
+        self.create_boundary(self.geo2, self.version)
+
+        self.hierarchy = self.create_hierarchy(
+            self.root, self.version
+        )
+        self.profile = ProfileFactory(geography_hierarchy=self.hierarchy)
+        self.category = CategoryFactory(profile=self.profile)
+        self.location = LocationFactory(category=self.category)
+
+    def tearDown(self):
+        cache.clear()
+
+    def create_boundary(self, geography, version):
+        return GeographyBoundaryFactory(
+            geography=geography, version=version
+        )
+
+    def create_hierarchy(self, root, version=None, **kwargs):
+        configuration = kwargs
+        if version:
+            configuration["default_version"] = version.name
+
+        return GeographyHierarchyFactory(
+            root_geography=root,
+            configuration=configuration
+        )
+
+    def create_pc(self, theme, order=0, label="pc label"):
+        return ProfileCategoryFactory(
+            profile=self.profile, category=self.category, theme=theme,
+            label=label, order=order
+        )
+
+    def create_multiple_profile_categories(self, theme, count=2):
+        pcs = []
+        for i in range(0, count):
+            pcs.append(
+                self.create_pc(theme, i, F"pc{i}_{theme.name}")
+            )
+        return pcs
+
+    def create_dataset(self, profile, version):
+        return DatasetFactory(
+            profile=profile, version=version
+        )
+
+    def create_datasetdata(
+            self, dataset, geography, data={},
+            create_groups=True
+        ):
+        obj = DatasetDataFactory(
+            dataset=dataset, geography=geography, data=data
+        )
+
+        if create_groups:
+            keys = data.keys()
+
+            for key in keys:
+                if key == "count":
+                    continue
+
+                group = Group.objects.filter(dataset=dataset, name=key).first()
+                if not group:
+                    group = GroupFactory(
+                        dataset=dataset, name="gender", subindicators=[],
+                        can_aggregate=True, can_filter=True
+                    )
+
+                if data[key] not in group.subindicators:
+                    subindicators = group.subindicators
+                    subindicators.append(data[key])
+                    group.subindicators = subindicators
+                    group.save()
+        return obj
+
+    def multiple_datasetdata(self, dataset, geography, datalist=[]):
+        objs = []
+        for data in datalist:
+            objs.append(self.create_datasetdata(dataset, geography, data))
+        return objs
```

Comment:
this method is redundant. it's always called with a single object.

also, its name is not a verb, so I have to read all its code to see what it's doing.

I suggest we remove this method for now and just call create_datasetdata directly for now, until we need something more fancy.